db/hints/internal/hint_sender.cc

```diff
@@ -80,9 +80,20 @@ future<> hint_sender::do_send_one_mutation(frozen_mutation_and_schema m, locator
     return futurize_invoke([this, m = std::move(m), ermp = std::move(ermp), &natural_endpoints] () mutable -> future<> {
         // The fact that we send with CL::ALL in both cases below ensures that new hints are not going
         // to be generated as a result of hints sending.
-        if (boost::range::find(natural_endpoints, end_point_key()) != natural_endpoints.end()) {
+        const auto maybe_addr = [&] () -> std::optional<gms::inet_address> {
+            const auto& tm = ermp->get_token_metadata();
+            if (auto maybe_ep = tm.get_endpoint_for_host_id_if_known(end_point_key())) {
```

Comment:
tangent observations:
  - since we look up the endpoint for every mutation, this is already safe for tablet migration while the node was down
  - but we need to extend the hold period of effective_replication_map_ptr until we get the response; it acts as a lock that prevents a tablet from migrating while the mutation is in-flight
 
Please follow up with extending the lifetime. Note that what we have now still works - @gusev-p will reject the mutation if it arrives with outdated topology. But it's better not to have to reject it.

Reply:
Another tangent: some coroutinizations would be helpful around here.